### ISSUE TITLE
Offline installation support/process

### ISSUE THREAD

**Issue by watsonb** [openshift/openshift-ansible#10163]
#### Description

I'm trying to conceive of a method to support the installation of OpenShift (Origin and Enterprise) on a disconnected/offline compute environment.  Our target environment will not have Internet access, so I'm attempting to use techniques I've employed in the past to cache needed artifacts on an Internet-connected environment, transfer that cache to the disconnected environment, and configure things in that environment to leverage the cache.  In essence, I'm bringing portions of the Internet with me.

To achieve this to date, I've leveraged a combination of custom caches, repo sync/mirror for OS packages, and Sonatype Nexus OSS (v3.13.0) as a caching transparent proxy.  However, I'm consistently running into errors with the `openshift-ansible` installer when trying to deploy a new cluster on an Internet-connected environment going through my Nexus proxy.

##### Version
```bash
(venv_ansible_2.6.2) [watsonb@ben-dev1 openshift-ansible]$ ansible --version
ansible 2.6.2
  config file = /home/watsonb/workspace/github/openshift-ansible/ansible.cfg
  configured module search path = [u'/home/watsonb/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/watsonb/venv_ansible_2.6.2/lib/python2.7/site-packages/ansible
  executable location = /home/watsonb/venv_ansible_2.6.2/bin/ansible
  python version = 2.7.5 (default, May  3 2017, 07:55:04) [GCC 4.8.5 20150623 (Red Hat 4.8.5-14)]

(venv_ansible_2.6.2) [watsonb@ben-dev1 openshift-ansible]$ git describe
openshift-ansible-3.9.44-1-5-gd85d6a9
```

##### Steps To Reproduce
1. run prerequisites.yml playbook (passes)
2. run deploy_cluster.yml playbook (fails)

##### Expected Results
OpenShift to install on specified Ansible inventory on an Internet-connected environment via a transparent Nexus caching proxy.

```
Example command and output or error messages
```

##### Observed Results
`docker_image_availability` check fails.  

```bash
1. Hosts:    ose3-test-infra1.grdn.pocnet.mil, ose3-test-infra2.grdn.pocnet.mil, ose3-test-master1.grdn.pocnet.mil, ose3-test-master2.grdn.pocnet.mil, ose3-test-master3.grdn.pocnet.mil, ose3-test-node1.grdn.pocnet.mil, ose3-test-node2.grdn.pocnet.mil
     Play:     OpenShift Health Checks
     Task:     Run health checks (install) - EL
     Message:  One or more checks failed
     Details:  check "docker_image_availability":
               One or more required container images are not available:
                   nexus03.grdn.pocnet.mil/openshift3/kubernetes:latest,
                   nexus03.grdn.pocnet.mil/openshift3/ose-deployer:v3.9.0,
                   nexus03.grdn.pocnet.mil/openshift3/ose-docker-registry:v3.9.0,
                   nexus03.grdn.pocnet.mil/openshift3/ose-haproxy-router:v3.9.0,
                   nexus03.grdn.pocnet.mil/openshift3/ose-pod:v3.9.0
               Checked with: skopeo inspect [--tls-verify=false] [--creds=<user>:<pass>] docker://<registry>/<image>
               Default registries searched: nexus03.grdn.pocnet.mil:19082, docker.io
```

When I disable this check, masters are unable to to pull images via Nexus.  Nexus logs show **404** errors indicating it cannot find a matching tag from Docker.io or https://registry.access.redhat.com/.

```bash
2018-09-19 12:48:53,482+0000 WARN  [qtp1580670561-2015]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.DockerProxyFacetImpl - Exception org.sonatype.nexus.repository.httpclient.RemoteBlockedIOException: Remote Auto Blocked until 2018-09-19T12:48:57.944Z checking remote for update, proxy repo docker-proxy failed to fetch v1/repositories/openshift3/ose-pod/tags/v3.9.0, content not in cache.
2018-09-19 12:48:53,483+0000 WARN  [qtp1580670561-2015]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.V1Handlers - Error: GET /v1/repositories/openshift3/ose-pod/tags/v3.9.0: 404 - org.sonatype.nexus.repository.docker.internal.V1Exception$TagNotFound: Tag not found
2018-09-19 12:48:57,280+0000 WARN  [qtp1580670561-2000]  anonymous org.sonatype.nexus.repository.docker.internal.V2Handlers - Is the remote url a valid docker endpoint? Remote host https://registry.access.redhat.com/ with path /v2/openshift3/ose-pod/manifests/v3.9.0 did not return the expected response. Error message: manifest unknown
2018-09-19 12:48:57,281+0000 WARN  [qtp1580670561-2000]  anonymous org.sonatype.nexus.repository.docker.internal.DockerProxyFacetImpl - Exception org.sonatype.nexus.repository.httpclient.RemoteBlockedIOException: Remote Auto Blocked until 2018-09-19T12:48:57.580Z checking remote for update, proxy repo aws-openshift failed to fetch v2/openshift3/ose-pod/manifests/v3.9.0, content not in cache.
2018-09-19 12:48:57,284+0000 WARN  [qtp1580670561-2000]  anonymous org.sonatype.nexus.repository.docker.internal.DockerProxyFacetImpl - Exception org.sonatype.nexus.repository.httpclient.RemoteBlockedIOException: Remote Auto Blocked until 2018-09-19T12:48:57.944Z checking remote for update, proxy repo docker-proxy failed to fetch v2/openshift3/ose-pod/manifests/v3.9.0, content not in cache.
2018-09-19 12:48:57,317+0000 WARN  [qtp1580670561-2015]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.DockerProxyFacetImpl - Exception org.sonatype.nexus.repository.httpclient.RemoteBlockedIOException: Remote Auto Blocked until 2018-09-19T12:48:57.580Z checking remote for update, proxy repo aws-openshift failed to fetch v1/repositories/openshift3/ose-pod/tags/v3.9.0, content not in cache.
2018-09-19 12:48:57,318+0000 WARN  [qtp1580670561-2015]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.V1Handlers - Error: GET /v1/repositories/openshift3/ose-pod/tags/v3.9.0: 404 - org.sonatype.nexus.repository.docker.internal.V1Exception$TagNotFound: Tag not found

GET /v1/repositories/openshift3/ose-pod/tags/v3.9.0: 404 - org.sonatype.nexus.repository.docker.internal.V1Exception$TagNotFound: Tag not found

2018-09-19 13:02:03,750+0000 WARN  [qtp1580670561-2041]  *UNKNOWN org.apache.http.impl.auth.HttpAuthenticator - TOKEN [complete=true] authentication error: Could not retrieve token from https://index.docker.io/. Status code: 404
2018-09-19 13:02:03,809+0000 WARN  [qtp1580670561-2041]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.DockerProxyFacetImpl - Exception org.sonatype.nexus.repository.proxy.ProxyServiceException: HTTP/1.1 401 UNAUTHORIZED checking remote for update, proxy repo docker-proxy failed to fetch v1/repositories/openshift3/ose-pod/tags/v3.9.0 with status line HTTP/1.1 401 UNAUTHORIZED, content not in cache.
2018-09-19 13:02:03,809+0000 WARN  [qtp1580670561-2041]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.V1Handlers - Error: GET /v1/repositories/openshift3/ose-pod/tags/v3.9.0: 404 - org.sonatype.nexus.repository.docker.internal.V1Exception$TagNotFound: Tag not found
```

##### Additional Information

My Ansible controller OS version information:
```bash
[watsonb@ben-dev1 openshift-ansible]$ cat /etc/redhat-release 
Red Hat Enterprise Linux Workstation release 7.4 (Maipo)
```
The target host OS version information:
```bash
[jenkins@ose3-test-master1 ~]$ cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.5 (Maipo)
```

My Ansible inventory:
```yaml
[masters]
ose3-test-master1.grdn.pocnet.mil
ose3-test-master2.grdn.pocnet.mil
ose3-test-master3.grdn.pocnet.mil

[etcd]
ose3-test-master1.grdn.pocnet.mil
ose3-test-master2.grdn.pocnet.mil
ose3-test-master3.grdn.pocnet.mil

[nodes]
ose3-test-master1.grdn.pocnet.mil openshift_node_group_name=node-config-master openshift_node_labels="{'region': 'master'}"
ose3-test-master2.grdn.pocnet.mil openshift_node_group_name=node-config-master openshift_node_labels="{'region': 'master'}"
ose3-test-master3.grdn.pocnet.mil openshift_node_group_name=node-config-master openshift_node_labels="{'region': 'master'}"
ose3-test-infra1.grdn.pocnet.mil openshift_node_group_name=node-config-infra openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
ose3-test-infra2.grdn.pocnet.mil openshift_node_group_name=node-config-infra openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
ose3-test-node1.grdn.pocnet.mil openshift_node_group_name=node-config-compute openshift_node_labels="{'region': 'apps'}"
ose3-test-node2.grdn.pocnet.mil openshift_node_group_name=node-config-compute openshift_node_labels="{'region': 'apps'}"

[nfs]
ose3-test-master1.grdn.pocnet.mil

[lb]
ose3-test-lb.grdn.pocnet.mil

[OSEv3:children]
masters
nodes
etcd
lb
nfs

[OSEv3:vars]
debug_level=2
ansible_user=jenkins
ansible_become=true
openshift_deployment_type=origin
openshift_release="3.9"
openshift_disable_check=disk_availability,memory_availability,docker_storage
openshift_enable_service_catalog=true
template_service_broker_install=true
openshift_hosted_infra_selector='region=infra'
openshift_hosted_router_selector='region=infra'
openshift_hosted_registry_selector='region=infra'
openshift_master_default_subdomain=apps-test.grdn.pocnet.mil
openshift_master_cluster_hostname=ose3-test-lb.grdn.pocnet.mil

# Cluster Image Source (registry) configuration
# openshift-enterprise default is 'registry.access.redhat.com/openshift3/ose-${component}:${version}'
# origin default is 'docker.io/openshift/origin-${component}:${version}'
oreg_url=nexus03.grdn.pocnet.mil:19082/openshift3/ose-${component}:${version}
openshift_examples_modify_imagestreams=true
openshift_docker_insecure_registries=nexus03.grdn.pocnet.mil:19082
openshift_docker_additional_registries=nexus03.grdn.pocnet.mil:19082
openshift_cockpit_deployer_prefix='nexus03.grdn.pocnet.mil/openshift3/'
openshift_web_console_prefix='nexus03.grdn.pocnet.mil/openshift3/ose-
openshift_service_catalog_image_prefix='nexus03.grdn.pocnet.mil/openshift3/ose-'
ansible_service_broker_image_prefix='nexus03.grdn.pocnet.mil/openshift3/ose-'
template_service_broker_prefix='nexus03.grdn.pocnet.mil/openshift3/ose-'
```

My Nexus information:

Nexus is configured with several Docker proxies (Docker.io, registry.access.redhat.com, etc.) placed into a docker-group.  The Nexus docker proxy is configured per instructions from Sonatype (HTTPS connectors configured, corresponding server certificates on Nexus and all servers pulling from Nexus, etc.).  I've enabled anonymous access to the server and have disabled authentication to the docker-group to eliminate such headaches for testing purposes.

From my desktop, I've configured `/etc/containers/registries.conf` as follows:

```ini
[jenkins@watsonb-dev1 ~]$ cat /etc/containers/registries.conf 
# Ansible managed

# This is a system-wide configuration file used to
# keep track of registries for various container backends.
# It adheres to TOML format and does not support recursive
# lists of registries.

# The default location for this configuration file is /etc/containers/registries.conf.

# The only valid categories are: 'registries.search', 'registries.insecure',
# and 'registries.block'.

[registries.search]
registries = ['nexus03.grdn.pocnet.mil:19082','registry.access.redhat.com','docker.io']

# If you need to access insecure registries, add the registry's fully-qualified name.
# An insecure registry is one that does not have a valid SSL certificate or only does HTTP.
[registries.insecure]
registries = ['nexus03.grdn.pocnet.mil:19082',]


# If you need to block pull access from a registry, uncomment the section below
# and add the registries fully-qualified name.
#
# Docker only
[registries.block]
registries = []
```

And I can do things like `docker pull hello-world`

```bash
[jenkins@watsonb-dev1 ~]$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
[jenkins@watsonb-dev1 ~]$ docker pull hello-world
Using default tag: latest
Trying to pull repository nexus03.grdn.pocnet.mil:19082/hello-world ... 
latest: Pulling from nexus03.grdn.pocnet.mil:19082/hello-world

d1725b59e92d: Pull complete 
Digest: sha256:1a6fd470b9ce10849be79e99529a88371dff60c60aab424c077007f6979b4812
```

And I can (sort of) pull openshift3 images if I specify the tag information correctly:

```bash
#
# tag v3.9.0 fails
#
[jenkins@watsonb-dev1 ~]$ docker pull openshift3/ose-pod:v3.9.0
Trying to pull repository nexus03.grdn.pocnet.mil:19082/openshift3/ose-pod ... 
Pulling repository nexus03.grdn.pocnet.mil:19082/openshift3/ose-pod
Trying to pull repository registry.access.redhat.com/openshift3/ose-pod ... 
Trying to pull repository docker.io/openshift3/ose-pod ... 
Pulling repository docker.io/openshift3/ose-pod
Trying to pull repository docker.io/openshift3/ose-pod ... 
Pulling repository docker.io/openshift3/ose-pod
Error: image openshift3/ose-pod:v3.9.0 not found

#corresponding Nexus log entry for failure

2018-09-20 11:16:21,406+0000 WARN  [qtp1580670561-2406]  anonymous org.sonatype.nexus.repository.docker.internal.V2Handlers - Is the remote url a valid docker endpoint? Remote host https://registry.access.redhat.com/ with path /v2/hello-world/blobs/sha256:d1725b59e92d6462c6d688ef028979cc6bb150762db99d18dddc7fa54b82b0ce did not return the expected response. Error message: blob unknown to registry
2018-09-20 11:16:21,701+0000 WARN  [qtp1580670561-2406]  anonymous org.sonatype.nexus.repository.docker.internal.DockerProxyFacetImpl - Exception com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: (org.apache.http.impl.io.EmptyInputStream); line: 1, column: 0] checking remote for update, proxy repo aws-openshift failed to fetch v2/hello-world/blobs/sha256:d1725b59e92d6462c6d688ef028979cc6bb150762db99d18dddc7fa54b82b0ce, content not in cache.
2018-09-20 11:16:21,701+0000 WARN  [qtp1580670561-3419]  anonymous org.sonatype.nexus.repository.docker.internal.DockerProxyFacetImpl - Exception com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: (org.apache.http.impl.io.EmptyInputStream); line: 1, column: 0] checking remote for update, proxy repo aws-openshift failed to fetch v2/hello-world/blobs/sha256:4ab4c602aa5eed5528a6620ff18a1dc4faef0e1ab3a5eddeddb410714478c67f, content not in cache.
2018-09-20 11:17:46,507+0000 WARN  [qtp1580670561-3430]  anonymous org.sonatype.nexus.repository.docker.internal.V2Handlers - Is the remote url a valid docker endpoint? Remote host https://registry.access.redhat.com/ with path /v2/openshift3/ose-pod/manifests/v3.9.0 did not return the expected response. Error message: manifest unknown
2018-09-20 11:17:47,123+0000 INFO  [qtp1580670561-3430]  anonymous org.sonatype.nexus.repository.httpclient.internal.HttpClientFacetImpl - Repository status for aws-openshift changed from AVAILABLE to UNAVAILABLE - reason Unauthorized for https://registry.reg-aws.openshift.com
2018-09-20 11:17:47,124+0000 WARN  [qtp1580670561-3430]  anonymous org.sonatype.nexus.repository.docker.internal.V2Handlers - Error: GET /v2/openshift3/ose-pod/manifests/v3.9.0: 401 - org.sonatype.nexus.repository.docker.internal.V2Exception: authentication required
2018-09-20 11:17:47,657+0000 WARN  [qtp1580670561-3430]  anonymous org.sonatype.nexus.repository.docker.internal.V2Handlers - Error: GET /v2/openshift3/ose-pod/manifests/v3.9.0: 401 - org.sonatype.nexus.repository.docker.internal.V2Exception: authentication required
2018-09-20 11:17:47,700+0000 WARN  [qtp1580670561-3430]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.V1Handlers - Error: GET /v1/repositories/openshift3/ose-pod/tags/v3.9.0: 404 - org.sonatype.nexus.repository.docker.internal.V1Exception$TagNotFound: Tag not found
2018-09-20 11:18:08,106+0000 WARN  [qtp1580670561-3430]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.auth.TokenScheme - Failed to retrieve docker token
org.apache.http.auth.AuthenticationException: Could not retrieve token from https://index.docker.io/. Status code: 404
2018-09-20 11:18:08,106+0000 WARN  [qtp1580670561-3430]  *UNKNOWN org.apache.http.impl.auth.HttpAuthenticator - TOKEN [complete=true] authentication error: Could not retrieve token from https://index.docker.io/. Status code: 404
2018-09-20 11:18:08,157+0000 WARN  [qtp1580670561-3430]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.DockerProxyFacetImpl - Exception org.sonatype.nexus.repository.proxy.ProxyServiceException: HTTP/1.1 401 UNAUTHORIZED checking remote for update, proxy repo docker-proxy failed to fetch v1/repositories/openshift3/ose-pod/tags/v3.9.0 with status line HTTP/1.1 401 UNAUTHORIZED, content not in cache.
2018-09-20 11:18:08,158+0000 WARN  [qtp1580670561-3430]  *UNKNOWN org.sonatype.nexus.repository.docker.internal.V1Handlers - Error: GET /v1/repositories/openshift3/ose-pod/tags/v3.9.0: 404 - org.sonatype.nexus.repository.docker.internal.V1Exception$TagNotFound: Tag not found


#
# tag v3.9 seems to work
#
[jenkins@watsonb-dev1 ~]$ docker pull openshift3/ose-pod:v3.9
Trying to pull repository nexus03.grdn.pocnet.mil:19082/openshift3/ose-pod ... 
Pulling repository nexus03.grdn.pocnet.mil:19082/openshift3/ose-pod
0b74b577f36a: Pull complete 
0ba22671cd98: Pull complete 
e2f4e4e239ef: Pull complete 
Status: Downloaded newer image for nexus03.grdn.pocnet.mil:19082/openshift3/ose-pod:v3.9
nexus03.grdn.pocnet.mil:19082/openshift3/ose-pod: this image was pulled from a legacy registry.  Important: This registry version will not be supported in future versions of docker.
[jenkins@watsonb-dev1 ~]$ docker images
REPOSITORY                                         TAG                 IMAGE ID            CREATED             SIZE
nexus03.grdn.pocnet.mil:19082/hello-world          latest              4ab4c602aa5e        12 days ago         1.84 kB
nexus03.grdn.pocnet.mil:19082/openshift3/ose-pod   v3.9                e26f0aabd752        3 weeks ago         214.2 MB
```

Has anyone out there accomplished something in the same spirit of what I'm trying to do here?  I understand this may not be a supported `openshift-ansible` use case, but I feel like I'm very close and there is some magic bit somewhere that needs to be flipped to make this work.

**Comment by watsonb** [comment]
This gets a little more bizarre.  I'm using the skopeo tool that the check uses on one of my masters.  I'm running it by hand from master1

```bash
#
# fails
#
skopeo --debug inspect --tls-verify=false docker://nexus03.grdn.pocnet.mil:19082/openshift3/ose-deployer:v3.9.0
FATA[0000] manifest unknown: manifest unknown 

#
# fails
#
skopeo --debug inspect --tls-verify=false docker://nexus03.grdn.pocnet.mil:19082/openshift3/ose-deployer:v3.9
FATA[0000] manifest unknown: manifest unknown 

#
# works
#
skopeo --debug inspect --tls-verify=false docker://nexus03.grdn.pocnet.mil:19082/openshift3/ose-deployer
DEBU[0001] GET https://nexus03.grdn.pocnet.mil:19082/v2/openshift3/ose-deployer/tags/list 
{
    "Name": "nexus03.grdn.pocnet.mil:19082/openshift3/ose-deployer",
...
}

#
# now this works because of the previous call that worked
#
skopeo --debug inspect --tls-verify=false docker://nexus03.grdn.pocnet.mil:19082/openshift3/ose-deployer:v3.9
DEBU[0001] GET https://nexus03.grdn.pocnet.mil:19082/v2/openshift3/ose-deployer/tags/list 
{
    "Name": "nexus03.grdn.pocnet.mil:19082/openshift3/ose-deployer",
...
}

#
# but tag v3.9.0 still won't work
#
skopeo --debug inspect --tls-verify=false docker://nexus03.grdn.pocnet.mil:19082/openshift3/ose-deployer:v3.9.0
FATA[0000] manifest unknown: manifest unknown 
```

**Comment by michaelgugino** [comment]
It seems your transparent proxy is the problem.  The installer works with alternative registries.  If you're failing to pull/use skopeo on the host, then obviously the installer won't be able to make that work.

**Comment by watsonb** [comment]
@michaelgugino understood.  I was hoping there was someone out there that had done something similar that could give a tip or hint.  I assumed because of the availability of the `oreg_url` variable and other related variables that this was "a thing" and am merely looking for a bit of validation that I'm not way out of bounds here.  I'll keep on poking around.

**Comment by watsonb** [comment]
curious, but when I do something like this:

```bash
skopeo --debug inspect --tls-verify=false docker://nexus03.grdn.pocnet.mil:19082/openshift3/ose-docker-registry:v3.9 | grep 3.9
```
why am I not seeing the tag "3.9.0" in the output?

```bash
        "v3.9",
        "v3.9.14",
        "v3.9.14-2",
        "v3.9.25",
        "v3.9.25-1",
        "v3.9.27",
        "v3.9.27-1",
        "v3.9.30",
        "v3.9.30-2",
        "v3.9.31",
        "v3.9.31-2",
        "v3.9.33",
        "v3.9.33-3",
        "v3.9.40",
        "v3.9.40-2",
        "v3.9.41",
        "v3.9.41-2"
        "url": "https://access.redhat.com/containers/#/registry.access.redhat.com/openshift3/ose-docker-registry/images/v3.9.41-2",
        "vcs-ref": "652088337453d9b3506d369d763077cbf91e3412",
        "version": "v3.9.41"
```

moreover, I thought with:
```yaml
openshift_deployment_type=origin
```
I'd be pulling from docker.io.

**Comment by michaelgugino** [comment]
Anything prefixed with 'ose-' is going to be an enterprise image.  For origin, you want
oreg_url="somehost.com/openshift/origin-${component}:${version}"

**Comment by watsonb** [comment]
@michaelgugino thanks for the tip.  I've adjusted my configuration as follows:

```yaml
#origin
oreg_url=nexus03.grdn.pocnet.mil:19082/openshift/origin-${component}:${version}
#origin
openshift_cockpit_deployer_prefix='nexus03.grdn.pocnet.mil:19082/cockpit/'
openshift_web_console_prefix='nexus03.grdn.pocnet.mil:19082/openshift/origin-
openshift_service_catalog_image_prefix='nexus03.grdn.pocnet.mil:19082/openshift/origin-'
ansible_service_broker_image_prefix='nexus03.grdn.pocnet.mil:19082/openshift/origin-'
template_service_broker_prefix='nexus03.grdn.pocnet.mil:19082/openshift/origin-
```

The docker image availability check still fails on kubernetes:latest, but I've confirmed on my masters by running skopeo directly:

```bash
skopeo --debug inspect --tls-verify=false docker://nexus03.grdn.pocnet.mil:19082/cockpit/kubernetes:latest
```

I got as far as the installer verifying the web console running before running into an error complaining about an invalid image tag.  The first time I ran it with the above config, I did not have the port number to my private registry set in the *_prefix variables, so I'm trying again with the port number specified.

Thanks again for the pointer.

**Comment by watsonb** [comment]
Just following up on this for posterity.  I eventually did get a disconnected/offline install to work.  Here's how I went about it.

I based my strategy on this document: https://docs.openshift.com/container-platform/3.10/install/disconnected_install.html

1. I started with a fresh Sonatype Nexus server install and setup docker proxy repos for docker.io and https://registry.access.redhat.com/ per Nexus documentation.
2. I had another machine with Docker (RedHat version) on it configured to use my Nexus box by editing /etc/containers/registries.conf and adding my Nexus server to the front of the list.
3. From this other machine, I performed a series of `docker pull <image>:<tag>` corresponding to the images/tags listed in the disconnected install article referenced above.  I actually wrote a little Ansible playbook to do this for me.  Also note that I switched from an `origin` installation to an `openshift-enterprise` installation and, hence, pulled the appropriate images to support that install.
4. After step 3 finished, my Nexus machine internally showed all of the requested images locally available.
5. I configured my Ansible inventory to include the following information:
```ini
openshift_deployment_type=openshift-enterprise
openshift_release="3.10"
openshift_image_tag=v3.10.45

#enterprise
oreg_url=<nexus_hostname>:<nexus_port>/openshift3/ose-${component}:${version}
#origin
#oreg_url=<nexus_hostname>:<nexus_port>/openshift/origin-${component}:${version}

# If oreg_url points to a registry other than registry.access.redhat.com we can
# modify image streams to point at that registry by setting the following to true

openshift_examples_modify_imagestreams=true

# Add insecure and blocked registries to global docker configuration
openshift_docker_insecure_registries=<nexus_hostname>:<nexus_port>

#openshift_docker_blocked_registries=registry.hacker.com

# You may also configure additional default registries for docker, however this
# is discouraged. Instead you should make use of fully qualified image names.
openshift_docker_additional_registries=<nexus_hostname>:<nexus_port>

#enterprise
openshift_cockpit_deployer_prefix='<nexus_hostname>/openshift3/'
openshift_web_console_prefix='<nexus_hostname>/openshift3/ose-'
openshift_service_catalog_image_prefix='<nexus_hostname>/openshift3/ose-'
ansible_service_broker_image_prefix='<nexus_hostname>/openshift3/ose-'
template_service_broker_prefix='<nexus_hostname>/openshift3/ose-'
```
6. I ran the `prerequisites.yml` and `deploy_cluster.yml` playbooks against fresh machines in my Ansible inventory and the install completed.

In other words, rather than having these images proxy/cache during installation time, I pre-populated my Nexus repo with the images I needed to support the install.  I can now use Nexus backup/restore functionality to take these images to disconnected networks and facilitate the install.

I'm closing this issue.
